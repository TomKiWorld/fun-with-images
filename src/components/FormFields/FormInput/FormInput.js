import React from 'react';

/**
 * Form input field
 * - Display short greet and amount of entries the user used the app
 * 
 * Required props:
 * - label => Displayed name
 * - name => For the is and label for
 * - type => Field type
 * - onChange => Function on input change
 * - onKeyUp => Function on key up - enter detection
 * Optional props:
 * - extraClass => For classlist 
 */
const FormInput = ({label, name, type, onChange, onKeyUp, extraClass}) => {
  const receivedClass= extraClass ?? '';
  let autocomplete = ''
  switch(name) {
    case 'password':
      autocomplete = 'current-password';
      break;
    case 'email-address':
      autocomplete = 'email';
      break;
    case 'name':
      autocomplete = 'name';
      break;
    default:
      autocomplete = '';
  }

  return (
    <div className='mt3'>
      <label className='db fw6 lh-copy f6' htmlFor={type}>{label}
        <input 
          onChange={onChange}
          onKeyUp={onKeyUp}
          className={`${receivedClass} pa2 input-reset ba b--black bg-transparent hover-bg-black hover-white w-100`} 
          type={type} 
          name={name} 
          id={name} 
          autoComplete={autocomplete}
        />
      </label>
    </div>
  );
}

export default FormInput;
