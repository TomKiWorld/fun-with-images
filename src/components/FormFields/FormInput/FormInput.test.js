import { shallow } from 'enzyme';
import React from 'react';
import FormInput from './FormInput';

describe('FormInput Component', () => {
  it('Expect to render FormInput Component', () => {
    expect(shallow(<FormInput />)).toMatchSnapshot();
  });

  it('Expect to renderFormInput Component as password', () => {
    const wrapper = shallow(<FormInput name={'password'}/>);
    const input = wrapper.find('input').at(0);
    expect(input.prop('name')).toBe('password');
  });

  it('Expect to render FormInput Component as email-address', () => {
    const wrapper = shallow(<FormInput name={'email-address'}/>);
    const input = wrapper.find('input').at(0);
    expect(input.prop('name')).toBe('email-address');
  });

  it('Expect to render FormInput Component as name', () => {
    const wrapper = shallow(<FormInput name={'name'}/>);
    const input = wrapper.find('input').at(0);
    expect(input.prop('name')).toBe('name');
  });

  it('Expect to render FormInput Component with placeholder', () => {
    const wrapper = shallow(<FormInput name={'name'} placeholder={'Guest'}/>);
    const input = wrapper.find('input').at(0);
    expect(input.prop('placeholder')).toBe('Guest');
  });
});
