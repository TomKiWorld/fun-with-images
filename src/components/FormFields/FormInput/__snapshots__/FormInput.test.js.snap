// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component Expect to render FormInput Component 1`] = `ShallowWrapper {}`;

exports[`FormInput Component Expect to render email-address FormInput Component 1`] = `ShallowWrapper {}`;

exports[`FormInput Component Expect to render name FormInput Component 1`] = `ShallowWrapper {}`;

exports[`FormInput Component Expect to render password FormInput Component 1`] = `ShallowWrapper {}`;
